File: typescript/23-common-chars.ts
// Caracteres en común — Fácil · Strings
// Cuenta los caracteres distintos compartidos por ambas cadenas, ignorando
// mayúsculas y minúsculas.
// Fuente: https://coding-challenges.dev/problems/typescript-caracteres-en-comun

export function commonChars(a: string, b: string): number {
  const charsA = new Set(a.toLowerCase());
  const charsB = new Set(b.toLowerCase());
  return [...charsA].filter((char) => charsB.has(char)).length;
}

File: python/23_common_chars.py
"""Caracteres en común — Fácil · Strings

Devuelve cuántos caracteres distintos aparecen en ambas cadenas, ignorando
mayúsculas y minúsculas.

Fuente: https://coding-challenges.dev/problems/python-caracteres-en-comun
"""


def common_chars(a: str, b: str) -> int:
    return len(set(a.lower()) & set(b.lower()))
